indico/modules/events/registration/controllers/management/reglists.py

```diff
@@ -185,3 +192,63 @@ class RHRegistrationEdit(RHManageRegistrationBase):
     def _process(self):
         return WPManageRegistration.render_template('management/registration_modify.html', self.event,
                                                     registration=self.registration)
+
+
+class RHRegistrationEmailRegistrants(RHManageRegFormBase):
+    """Send email to selected registrants"""
+
+    def _checkParams(self, params):
+        self._doNotSanitizeFields.append('from_address')
+        RHManageRegFormBase._checkParams(self, params)
+
+    def _send_emails(self, form):
+        ids = set(request.form.getlist('registration_ids'))
+        registrations = (Registration
```

Comment:
I guess this is the same logic for deletion, emailing, etc? Maybe a `RHRegistrantsActionBase` assigning `self.registrants` in checkParams might be a good idea?